datafusion/optimizer/src/scalar_subquery_to_join.rs

```diff
@@ -510,6 +494,7 @@ mod tests {
             vec![
                 Arc::new(ScalarSubqueryToJoin::new()),
                 Arc::new(ExtractEquijoinPredicate::new()),
+                Arc::new(EliminateCrossJoin::new()),
```

Comment:
I think these tests are getting a little hard to understand because they use a subset of the optimizer passes that is not used for actual queries, bit is more than pass that is defined in this module. 

I recommend that in a follow on PR we either
1. remove all optimizer passes in the tests in this file other than `ScalarSubqueryToJoin` (so the tests here are focused on the code in this module)
2. Move any query plans that we want to see the final plan after all rewrites to the end to end tests (ideally sqllogictest) files

Reply:
Sure, I will do this in a following PR.